File: example-filters/filter-passthrough.rb
#! /usr/bin/env ruby

# A trivial 6jack filter that receives messages and echoes them.
# All messages are printed to stderr by the way.

require "msgpack"
require "awesome_print"

pac = MessagePack::Unpacker.new
loop do
  begin
    data = STDIN.readpartial(65536)
  rescue EOFError
    break
  end
  pac.feed(data)
  pac.each do |obj|
    warn obj.awesome_inspect
    STDOUT.write obj.to_msgpack
    STDOUT.flush
  end
end

File: example-filters/filter-passthrough.php
#! /usr/bin/env php
<?
    $unpacker = new MessagePackUnpacker();
    $buffer = "";
    $nread = 0;
    $false = 0;
    
    while (1) {
        $data = fread(STDIN, 65536);
        if ($data === FALSE || feof(STDIN)) break;
        
        $buffer = $buffer . $data;
        while(true) {
            if($unpacker->execute($buffer, $nread)){
                $msg = $unpacker->data();

                fwrite(STDERR, print_r($msg, true));
                
                fwrite(STDOUT, msgpack_pack($msg));
                $unpacker->reset();
                $buffer = substr($buffer, $nread);
                $nread = 0;

                if (!empty($buffer)){
                    continue;
                }
            }
            break;
        }
    }
  
?>
